src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,733 @@
+     * @return pendingMagnitude the pending allocations for each operatorSet
+     * @return timestamps the timestamps for each pending allocation
+     */
+    function getPendingAllocations(
+        address operator,
+        IStrategy strategy,
+        OperatorSet[] calldata operatorSets
+    ) external view returns (uint64[] memory, uint32[] memory) {
+        uint64[] memory pendingMagnitude = new uint64[](operatorSets.length);
+        uint32[] memory timestamps = new uint32[](operatorSets.length);
+        for (uint256 i = 0; i < operatorSets.length; ++i) {
+            // We use latestSnapshot to get the latest pending allocation for an operatorSet.
+            (bool exists, uint32 key, uint224 value) =
+                _magnitudeUpdate[operator][strategy][_encodeOperatorSet(operatorSets[i])].latestSnapshot();
+            if (exists) {
+                if (key > block.timestamp) {
+                    pendingMagnitude[i] = uint64(value);
+                    timestamps[i] = key;
+                } else {
+                    pendingMagnitude[i] = 0;
+                    timestamps[i] = 0;
+                }
+            }
+        }
+        return (pendingMagnitude, timestamps);
+    }
+
+    /**
+     * @notice Returns the pending deallocations of an operator for a given strategy and operatorSets.
+     * One of the assumptions here is we don't allow more than one pending deallocation for an operatorSet at a time.
+     * If that changes, we would need to change this function to return all pending deallocations for an operatorSet.
+     * @param operator the operator to get the pending deallocations for
+     * @param strategy the strategy to get the pending deallocations for
+     * @param operatorSets the operatorSets to get the pending deallocations for
+     * @return pendingMagnitudes the latest pending deallocation
+     */
+    function getPendingDeallocations(
+        address operator,
+        IStrategy strategy,
+        OperatorSet[] calldata operatorSets
+    ) external view returns (PendingFreeMagnitude[] memory) {
+        PendingFreeMagnitude[] memory pendingMagnitudes = new PendingFreeMagnitude[](operatorSets.length);
+        for (uint256 i = 0; i < operatorSets.length; ++i) {
+            uint256[] storage deallocationIndices =
+                _queuedDeallocationIndices[operator][strategy][_encodeOperatorSet(operatorSets[i])];
+
+            if (deallocationIndices.length > 0) {
+                uint256 deallocationIndex = deallocationIndices[deallocationIndices.length - 1];
+                PendingFreeMagnitude storage latestPendingMagnitude =
+                    _pendingFreeMagnitude[operator][strategy][deallocationIndex];
+                if (latestPendingMagnitude.completableTimestamp > block.timestamp) {
+                    pendingMagnitudes[i] = latestPendingMagnitude;
+                }
+            }
+        }
+        return pendingMagnitudes;
```

Comment:
so if no deallocation is pending this will return 'empty' `PendingFreeMagnitude` structs. might be worth specifying this behavior (at least in the natspec)?